protocol/zipper/zipper.go

```diff
@@ -0,0 +1,86 @@
+package zipper
+
+import (
+	"bytes"
+	"compress/gzip"
+	"io"
+	"net/http"
+	"sync"
+	"sync/atomic"
+
+	"github.com/signalfx/golib/v3/datapoint"
+	"github.com/signalfx/golib/v3/log"
+	"github.com/signalfx/golib/v3/sfxclient"
+)
+
+// ReadZipper creates a Pool that contains previously used Readers and can create new ones if we run out.
+type ReadZipper struct {
+	zippers   sync.Pool
+	Log       log.Logger
+	NewCount  int64
+	HitCount  int64
+	MissCount int64
+	ErrCount  int64
+}
+
+// GzipHandler transparently decodes your possibly gzipped request
+func (z *ReadZipper) GzipHandler(h http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		var err error
+		if r.Header.Get("Content-Encoding") == "gzip" {
```

Comment:
In the case of this method getting called when the encoding is not gzip, the miss count will be incremented. Is that what we want?

Reply:
Apparently it's how it's been working, I'd keep it as is